app/controllers/api/v1/customers_controller.rb

```diff
@@ -54,13 +68,28 @@ def customer
         @customer ||= Customer.find(params[:id])
       end
 
+      def set_display_customer_balance
+        val = params[:display_customer_balance]
+        return if val.nil?
+
+        unless val.in?(["true", "false", "1", "0"])
+          invalid_query_parameter(:display_customer_balance, :unprocessable_entity, "Not a boolean")
+        end
+        @display_customer_balance = ActiveModel::Type::Boolean.new.cast(val)
```

Comment:
Most of this could be in a re-usable method. The text "Not a boolean" should be translated via `I18n`.

Reply:
Fixed with `Api::V1::QueryParam` and related translations, now for `extra_fields` instead of `boolean`